core/src/main/java/nl/nn/adapterframework/jms/PushingJmsListener.java

```diff
@@ -140,12 +146,18 @@ public void close() {
 
 
 	@Override
-	public IListenerConnector<javax.jms.Message> getListenerPortConnector() {
+	public IListenerConnector<Message> getListenerPortConnector() {
 		return jmsConnector;
 	}
 
-
-
+	@Override
+	public RawMessageWrapper<Message> wrapRawMessage(Message rawMessage, Map<String, Object> threadContext) throws ListenerException {
+		Map<String, Object> messageContext = populateContextFromMessage(rawMessage);
+		String id = (String) messageContext.get(PipeLineSession.MESSAGE_ID_KEY);
+		String cid = (String) messageContext.get(PipeLineSession.CORRELATION_ID_KEY);
+		threadContext.putAll(messageContext);
```

Comment:
De MessageContext hoort bij een Message en moet dus niet in de ThreadContext komen.

Reply:
Ik hernoem het nu naar `Map<String, Object> messageProperties = extractMessageProperties(rawMessage);`, de naam `threadContext` blijkt eigenlijk al helemaal fout te zijn. Ik heb dit nu veranderd in `PipeLineSession session`, want werd overal nu aangeroepen met de session.